rust/system/src/execution/orchestrator.rs

```diff
@@ -1,12 +1,54 @@
+use super::Operator;
 use crate::{ChannelError, Component, ComponentContext, ComponentHandle, PanicError, System};
+use crate::{Dispatcher, TaskMessage};
 use async_trait::async_trait;
 use chroma_error::ChromaError;
 use core::fmt::Debug;
+use parking_lot::Mutex;
 use std::any::type_name;
+use thiserror::Error;
 use tokio::sync::oneshot::{self, error::RecvError, Sender};
+use tracing::Instrument;
 use tracing::Span;
 
-use crate::{Dispatcher, TaskMessage};
+#[derive(Debug)]
+struct WrappedOrchestratorOperator<O: Orchestrator>(Mutex<Option<O>>, System, Span);
```

Comment:
I prefer a named struct here. Also think `OrchestratorOperator` is a better name.